Bug 1693489 -  Update Glean SDK development and Glean book front page content (#1528)
mozilla/glean: docs/user/README.md

# Introduction

> Glean is a modern approach for a Telemetry library and is part of the [Glean project](https://docs.telemetry.mozilla.org/concepts/glean/glean.html).

Reviewer: While we're here make it lowercase telemetry:
```suggestion
Glean is a modern approach for a telemetry library and is part of the [Glean project](https://docs.telemetry.mozilla.org/concepts/glean/glean.html).
```

## Contact

To contact the Glean team you can:

- Find us in the [#glean channel on chat.mozilla.org](https://chat.mozilla.org/#/room/#glean:mozilla.org).
- Send an email to *glean-team@mozilla.com*.

> - The Glean SDK team is: *:janerik*, *:dexter*, *:travis*, *:mdroettboom*, *:gfritzsche*, *:chutten*, *:brizental*.

Reviewer: Should we update this? Georg's out for a while now :(

Reviewer 2: Or we can move his name to a "contributors" or "emeritus" section...

Author: "alumni" maybe? 

That would not fit in the `# Contact` section though, IMO. How about we remove his name here and file a bug for adding a `contributors` section somewhere else? Maybe the Github README? We could maybe even automate that based on commit history or something. Outside contributors might appreciate that :D

Reviewer: Let's leave it and tackle that separately.

> ![Glean logo](glean.jpeg)

Reviewer: What was the reason for moving the logo down here?
IMO it was good as the big splash to start with.

Plus now the contact info is front and center and I'm not sure that's the best way

Author: Yeah, I had put the contact before the `## Sections` title, but I was torn... I can move it back there, or somewhere else. The truth is I didn't know where to put it :D

Reviewer 3: Side-note: do we want the logo with a white or transparent background to be shown here?

Author: > Side-note: do we want the logo with a white or transparent background to be shown here?

I thought about this. The black adds a bit of drama, which I appreciate. IMO, we should keep it black.

Author: (I moved the `Contact` section to right before the `License` section in this commit https://github.com/mozilla/glean/pull/1528/commits/e45e44225e54cf570489499c1c9928d2dbab72eb, still not 100% secure about its positioning)

Unless clearly stated otherwise, regard the text in this book as valid for both clients and all the supported programming languages and environments.

### [The Glean SDK](https://github.com/mozilla/glean)

The Glean SDK is an implementation of Glean in Rust, with language bindings for **Kotlin**, **Python**, **Rust** and **Swift**.

For development documentation on the `Glean SDK`, refer to [the Glean SDK development book](https://mozilla.github.io/glean/).

To report issues or request changes on the Glean SDK, file a bug in [Bugzilla in Data Platform & Tools :: Glean: SDK](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data+Platform+and+Tools&component=Glean%3A+SDK&priority=P3&status_whiteboard=%5Btelemetry%3Aglean-rs%3Am%3F%5D).

### [Glean.js](https://github.com/mozilla/glean.js)

Glean.js is an implementation of Glean in **Javascript**. Currently, it only has support for usage in web extensions.

For development documentation on `Glean.js`, refer to [the Glean.js development documentation](https://github.com/mozilla/glean.js/tree/main/docs).

> To report issues or request changes on the Glean SDK, file a bug in [Bugzilla in Data Platform & Tools :: Glean.js](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data+Platform+and+Tools&component=Glean.js&priority=P4&status_whiteboard=%5Btelemetry%3Aglean-js%3Am%3F%5D).

Reviewer: ```suggestion
To report issues or request changes on the Glean SDK, file a bug in [Bugzilla in Data Platform & Tools :: Glean.js][gleanjs-bugs].
```

Author: See https://github.com/mozilla/glean/pull/1528#discussion_r585379503,do you know a workaround for the link checker issue?

> > **Note** Glean.js is still on the MVP stage and does not provide all the features the Glean SDK does. Feature parity will be worked on after initial validation. Do not hesitate to file bugs using the link above if you want to use Glean.js and is missing some key Glean feature.

Reviewer: ```suggestion
> **Note** Glean.js is still in development and does not provide all the features the Glean SDK does. Feature parity will be worked on after initial validation. Do not hesitate to [file a bug][gleanjs-bugs] if you want to use Glean.js and is missing some key Glean feature.

[gleanjs-bugs]: https://bugzilla.mozilla.org/enter_bug.cgi?product=Data+Platform+and+Tools&component=Glean.js&priority=P4&status_whiteboard=%5Btelemetry%3Aglean-js%3Am%3F%5D
```

Author: I seems the link checked is not to happy about the link alias :/

## Sections

### [Using Glean](./user/index.html)

> If you want to start using Glean to report data then this is the section you should read. It explains the first steps of integrating Glean into your project, choosing the right metric type for you, debugging products that use Glean and even Glean's internal error reporting mechanism.

Reviewer: I think I would turns this around slightly:

```suggestion
In this chapter we describe how to use the Glean SDK in your own libraries and applications.
It explains the first steps of integrating Glean into your project, choosing the right metric type for you, debugging products that use Glean and Glean's built-in error reporting mechanism.

If you want to start using Glean to report data then this is the section you should read.
```

Reviewer: Works for me!

One last nitpick: You use `section` for the other parts so I think it'd be better to also say "In this section ..." here. Sorry, that was my mistake to suggest "chapter"

For development documentation on the `Glean SDK`,
refer to [the Glean SDK development book](https://mozilla.github.io/glean/).

To report issues or request changes on the Glean SDK,
file a bug in [Bugzilla in Data Platform & Tools :: Glean: SDK](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data+Platform+and+Tools&component=Glean%3A+SDK&priority=P3&status_whiteboard=%5Btelemetry%3Aglean-rs%3Am%3F%5D).

### [Glean.js](https://github.com/mozilla/glean.js)

Glean.js is an implementation of Glean in **Javascript**. Currently, it only has support
for usage in web extensions.

For development documentation on `Glean.js`,
refer to [the Glean.js development documentation](https://github.com/mozilla/glean.js/tree/main/docs).

To report issues or request changes on Glean.js,
file a bug in [Bugzilla in Data Platform & Tools :: Glean.js][gleanjs-bugs].

> **Note** Glean.js is still in development and does not provide all the features the Glean SDK does.
> Feature parity will be worked on after initial validation. Do not hesitate to [file a bug][gleanjs-bugs]
> if you want to use Glean.js and is missing some key Glean feature.
## Sections

### [Using Glean](./user/index.html)

> In this chapter we describe how to use Glean in your own libraries and applications.

Reviewer: ```suggestion
In this section we describe how to use Glean in your own libraries and applications.
```

to make it consistent with the next parts. Sorry for suggesting "chapter" to begin with.
